docs/PROI/report.tex

```diff
@@ -45,6 +45,90 @@ \subsubsection{Stretch Goals}
+Die E-Mail wäre hierbei das einzigartigste Merkmal. Über Scopes in der OAuth Anfrage kann diese auch mitgeliefert werden.
+Das setzt aber voraus, dass Nutzer immer die gleiche E-Mail nutzen, was nicht immer der Fall ist.
+Man könnte dem (bereits eingeloggten) User die Möglichkeit geben, weitere Accounts zu diesem hinzuzufügen, sodass diese Accounts den gleichen Nutzer einloggen.
+Bei diesem Ansatz sollten dann die jeweiligen verbundenen Accounts zu einer unique User-ID verbunden werden.
+
+\subsubsection{Musik Artwork laden}
+Sollte kein Problem sein, da alle Anfragen ein Array von Bildern enthalten \footnote{https://stackoverflow.com/questions/10123804/retrieve-cover-artwork-using-spotify-api}
+
+\subsubsection{Eigene Bilder hochladen}
+Eigene Bilder hochzuladen sollte möglich sein. In unserem Kontext mit Vue.js und Node.js würde das Plugin \textit{vue-picture-input} helfen.
+Damit könnte mit einem Axios Post an das Backend das Bild gesendet werden. \footnote{https://www.digitalocean.com/community/tutorials/vuejs-uploading-vue-picture-input}
+
+\subsubsection{Spotify connect Lautsprecher auswählen}
+Das Auswählen von einem spezifischen Spotify Connect Lautsprecher ist möglich.
+Über einen API Call an die Spotify API mit dem Endpunkt \textit{/v1""/me""/player""/device} liefert eine Liste von allen verbundenen Geräten.
+Es kann auch der letzte aktive Lautsprecher genutzt werden. Dieser hat bei \textit{is\_active} den Wert \textit{true}. \footnote{https://developer.spotify.com/documentation/web-api/guides/using-connect-web-api/}
+
+\subsubsection{Spotify connect Lautsprecher speichern}
+Die Liste von verbundenen Geräten, die über den API Call an die Spotify API erhalten wird, enthält auch einen unique Eintrag \textit{id}. Dieser kann gespeichert werden.
+
+\subsubsection{In Cloud UI die jeweilige Box auswählen}
+In dem Szenario, dass wir nur ein Cloud UI bieten für alle Boxen bieten, muss es auch eine Möglichkeit geben, die Boxen auszuwählen, die gerade konfiguriert werden soll.
+Jede Box könnte ein QR-Code erhalten, der beim Einloggen gescannt werden muss. Je nachdem kann diese Box dann im Account hinterlegt werden oder muss bei jedem Login gescannt werden.
+
+\subsubsection{Boxdaten über Cloud UI ändern}
+Dafür müsste eine API gebaut werden, damit der jeweilige Microcontroller der Box mit dem Cloud UI kommunizieren kann.
+
+\subsubsection{Unterstützung von Youtube Music}
+Eine Umsetzung könnte sich als umständlich erweisen, da es bisher noch keine dedizierte Youtube Music API gibt. In der bisherigen API gibt nur die Möglichkeit, nach Youtube Videos zu suchen \footnote{https://developers.google.com/youtube/v3/}. Dazu kommt auch das Problem, dass ein externes Wiedergabegerät wie bei Spotify Connect derzeitig in der Form noch nicht existiert.
+\subsubsection{Unterstützung von Apple Music}
+Ist möglich. Apple Music bietet hier mit deren MusicKit JS\footnote{https://developer.apple.com/documentation/musickitjs/} eine Möglichkeit hierfür.
```

Comment:
Ein paar mehr Infos, ist das unter Linux / auf dem Pi machbar, wären hier sicherlich gut.

```suggestion
Apple Music bietet hier mit deren MusicKit JS\footnote{https://developer.apple.com/documentation/musickitjs/} eine Möglichkeit, um Musik abzuspielen.
```

Reply:
resolved in commit 0ee0602e5751e2c119ce92df11403e7bdf215228